ls1intum/Artemis
# `General`: Improve LTI configuration page

### Checklist
#### Generaltest on the test servers. -->
#### Server
#### Client

### Motivation and Context

The LTI configuration page isn't very clear for instructors, specifically which fields can and can't be configured. It's also not very consistent with the rest of Artemis.

### Description

Improves the LTI configuration page by making it clearer which fields are required/should be configured in a certain way, and which fields the instructor is free to configure. Tooltips are also added for the most "confusing" fields to help the instructor understand their meaning. The instructor can change the "Custom parameters" and then copy the configuration and paste it into the LMS. Also makes the fields more consistent with the rest of Artemis. The alert shown when LTI isn't configured is changed from an access not allowed to a custom alert.

### Steps for Testing
Prerequisites:
- 1 Instructor

Test the improved page - can only be tested in ts1 (since LTI is only configured in ts1)

1. Log in to Artemis
2. Navigate to Course Management
3. Create or navigate to an online course and create or navigate to an exercise within the course
4. Navigate to the exercise details page, and then navigate to "LTI Configuration"
5. Check the changes in the page, verify that the tooltips are helpful and that the translations are correct
7. Verify that you can edit the custom parameters and copy them

Test the new alert - can't be tested in ts1, can be tested in another test server or locally

1. Log in to Artemis
2. Navigate to Course Management
3. Create or navigate to an online course and create or navigate to an exercise within the course
4. Navigate to the exercise details page, and then navigate to "LTI Configuration"
5. Verify that you get the alert "LTI is not supported on this Artemis instance" instead of a "Access not allowed"

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage

No changes

### Screenshots

**Before**

![lti_configuration_old](https://user-images.githubusercontent.com/11006274/188816073-b04576aa-2093-4c48-aa3a-389f7cf43823.png)

![lti_access_not_allowed](https://user-images.githubusercontent.com/11006274/188815484-b789c899-4b6b-491c-b59d-c0cbc1fa42a8.png)


**After**
![improved-lti-configuration-page](https://user-images.githubusercontent.com/11006274/188848909-32540407-421d-4634-af29-41311c2c5936.png)

![lti_not_supported](https://user-images.githubusercontent.com/11006274/188812922-1ab19562-6db6-4e48-b757-32c1e5a5cc8c.png)
